api/PayrollProcessor.Data.Persistence/Infrastructure/Clients/QueueClientFactory.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+using Ardalis.GuardClauses;
+using Azure;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using Newtonsoft.Json;
+
+namespace PayrollProcessor.Data.Persistence.Infrastructure.Clients
+{
+    public interface IMessage
+    {
+        string Source { get; set; }
+    }
+
+    public static class QueueMessageBuilder
```

Comment:
I'm not completely happy with this way of integrating with sending Queue messages, but I don't think it makes sense to try to come up with something better until we use these more and find problems with them.